broker/src/main/java/io/camunda/zeebe/broker/system/partitions/PartitionTransition.java

```diff
@@ -7,32 +7,14 @@
  */
 package io.camunda.zeebe.broker.system.partitions;
 
+import io.camunda.zeebe.protocol.record.PartitionRole;
 import io.camunda.zeebe.util.sched.future.ActorFuture;
 
 public interface PartitionTransition {
 
-  /**
-   * Transitions to follower asynchronously by closing the current partition's components and
-   * opening a follower partition.
-   *
-   * @param currentTerm the current term on which the transition happens
-   * @return an ActorFuture to be completed when the transition is complete
-   */
-  ActorFuture<Void> toFollower(final long currentTerm);
-
-  /**
-   * Transitions to leader asynchronously by closing the current partition's components and opening
-   * a leader partition.
-   *
-   * @param currentTerm the current term on which the transition happens
-   * @return an ActorFuture to be completed when the transition is complete
-   */
-  ActorFuture<Void> toLeader(final long currentTerm);
-
-  /**
-   * Closes the current partition's components asynchronously.
-   *
-   * @return an ActorFuture completed when the transition is complete
-   */
-  ActorFuture<Void> toInactive();
+  ActorFuture<PartitionContext> toRole(
```

Comment:
I think I would like it more if we have defined here the transition which can happen.

Like: LeaderToFollower, FollowerToLeader, etc. This allows us in the implementation to concentrate on the key things and reduced boiler plate. We don't need to have always the switch etc.